### ISSUE TITLE
Problem training pytorch_networks/masks subnet

### ISSUE THREAD

**Issue by gajdosech2** [Shreeyak/cleargrasp#42]
Hello,

first off, thank you very much for your work. 
I tried retraining the segmentation network for higher resolution (800x600), however, I have failed to make the training work, even for the default 256x256 resolution. Specifically, I have downloaded your training dataset and tried to continue training from your `checkpoint_mask.pth`. However, these are the results after the first epoch:

> Train Epoch Loss: 11751.5604
> Train mIoU: 0.0000
> 
> Validation Epoch Loss: 704898596864.0000
> Validation mIoU: 0.0000
> 
> Test Synthetic Epoch Loss: 3180020034638.7690
> Test Synthetic mIoU: 0.0000

Visually, the network produces just black frames, whereas your trained checkpoint works as expected. I have tried a similar approach for the surface normal prediction network (again using your pre-trained weights and just increasing the resolution) and there it worked fine.

I believe there is some problem with loss function / training code for this `mask` network, but I have failed to pinpoint it in the code. 

I would appreciate any help, thank you.

**Comment by astrongtiger** [comment]
我也遇到了一样的问题，loss很大，iou为0，一直无法解决

**Comment by Zhh-doyourself** [comment]
您好！我也遇到了同样的问题，Epoch 损失值特别大，到达了几万，另外 mIoU一直为0.0000，请问您现在解决了吗？

**Comment by gajdosech2** [comment]
@Zhh-doyourself I noticed your message in my other repo, unfortunately no, this issue was not solved.

**Comment by Zhh-doyourself** [comment]
OK，thank you! what a pity, do you know any other solution? Because it's my senior thesis, and I have to solve it，looking forward to hearing from you.

---Original---
From: ***@***.***&gt;
Date: Wed, Feb 26, 2025 15:50 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [Shreeyak/cleargrasp] Problem training pytorch_networks/maskssubnet (Issue #42)


   
@Zhh-doyourself I noticed your message in my other repo, unfortunately no, this issue was not solved.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you were mentioned.Message ID: ***@***.***&gt;
  gajdosech2 left a comment (Shreeyak/cleargrasp#42)
 
@Zhh-doyourself I noticed your message in my other repo, unfortunately no, this issue was not solved.

**Comment by gajdosech2** [comment]
Well you can either try to fix the code or replace the network for mask segmentation with something else, something like a UNet.

**Comment by Zhh-doyourself** [comment]
> 那么您可以尝试修复代码或者用其他东西（例如 UNet）替换用于掩码分割的网络。

好的，谢谢，我去试一试
